In [1]:
import numpy as np
import pandas as pd
import tables as tb
import sys
eps = sys.float_info.epsilon

In [2]:
def read_event(fname, datid, table='Voxels', group='DATASET'):
    with tb.open_file(fname) as h5in:
        return pd.DataFrame.from_records(h5in.root[group][table].read_where('dataset_id==datid'))

In [5]:
dataset_path = '/home/mmkekic/MC_dataset/new_data/train_dataset_5mm_all.h5'

In [6]:
dataset_id = 24730
event = read_event(dataset_path, dataset_id, table='Voxels')


In [7]:
def class_ratios(event_id, data_path):
    event = read_event(data_path, event_id, table='Voxels')
    count = np.bincount(event['segclass'], minlength = 3)
    inv = 1./(count + eps) #in case some class has 0 counts
    return inv/sum(inv)

In [16]:
n = 1000
mean = np.zeros(3)

for i in range(n):
    #ev_id = np.random.randint(60000)
    ratio = class_ratios(i, dataset_path)
    mean += ratio
print('Mean:', mean/n)

Mean: [0.39100247 0.05181371 0.55718382]


Para 5mm:
    
[0.38890177 0.05312072 0.5579775 ] (1000 ev)

[0.38925264 0.05114205 0.55960531] (5000 ev)

[0.39121372 0.05125287 0.55753341] (10000 ev)

[0.39100247 0.05181371 0.55718382] (50000 ev)

Para 1mm:

[0.54925774 0.06549303 0.44924923]

[0.54041125 0.06872425 0.4628645 ]

[0.52914039 0.06320249 0.46365712]

[0.53607411 0.06864869 0.4632772 ]